sdk/core/azure-core/src/environment_log_level_listener.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#if defined(AZ_NO_ENV_LOGGER)
+#include "azure/core/platform.hpp"
+
+#if defined(AZ_PLATFORM_WINDOWS)
+#if !defined(WIN32_LEAN_AND_MEAN)
+#define WIN32_LEAN_AND_MEAN
+#endif
+#if !defined(NOMINMAX)
+#define NOMINMAX
+#endif
+
+#include <windows.h>
+#endif
+#endif
+
+#if !defined(AZ_NO_ENV_LOGGER) \
+    && (!defined(WINAPI_PARTITION_DESKTOP) \
+        || WINAPI_PARTITION_DESKTOP) // See azure/core/platform.hpp for explanation.
+
+#include "environment_log_level_listener_private.hpp"
+
+#include "azure/core/datetime.hpp"
+#include "azure/core/internal/strings.hpp"
+
+#include <iostream>
+#include <string>
+
+using namespace Azure::Core;
+using namespace Azure::Core::Details;
+
+namespace {
+Logger::Level const* GetEnvironmentLogLevel()
+{
+  static Logger::Level* envLogLevelPtr = nullptr;
+
+  static bool initialized = false;
+  if (!initialized)
+  {
+    initialized = true;
+
+#if defined(_MSC_VER)
+#pragma warning(push)
+// warning C4996: 'getenv': This function or variable may be unsafe. Consider using _dupenv_s
+// instead.
+#pragma warning(disable : 4996)
+#endif
+    auto envVar = std::getenv("AZURE_LOG_LEVEL");
+#if defined(_MSC_VER)
+#pragma warning(pop)
+#endif
+
+    if (envVar)
+    {
+      auto const logLevelStr = Internal::Strings::ToLower(envVar);
+
+      // See https://github.com/Azure/azure-sdk-for-java/wiki/Logging-with-Azure-SDK
+      // And
+      // https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/util/logging/LogLevel.java
+
+      static Logger::Level envLogLevel = {};
+      envLogLevelPtr = &envLogLevel;
+
+      if (logLevelStr == "error" || logLevelStr == "err" || logLevelStr == "4")
+      {
+        envLogLevel = Logger::Level::Error;
+      }
+      else if (logLevelStr == "warning" || logLevelStr == "warn" || logLevelStr == "3")
+      {
+        envLogLevel = Logger::Level::Warning;
+      }
+      else if (
+          logLevelStr == "informational" || logLevelStr == "info" || logLevelStr == "information"
+          || logLevelStr == "2")
+      {
+        envLogLevel = Logger::Level::Informational;
+      }
+      else if (logLevelStr == "verbose" || logLevelStr == "debug" || logLevelStr == "1")
+      {
+        envLogLevel = Logger::Level::Verbose;
+      }
+      else
+      {
+        envLogLevelPtr = nullptr;
+      }
+    }
+  }
+
+  return envLogLevelPtr;
+}
+
+// Log level textual representation, including space padding, matches slf4j and log4net.
+std::string const ErrorText = "ERROR";
+std::string const WarningText = "WARN ";
+std::string const InformationalText = "INFO ";
+std::string const VerboseText = "DEBUG";
+std::string const UnknownText = "?????";
```

Comment:
what if we move this as `static constexpr char[]`

Reply:
Then we can't return `std::string const&` from the inline functin below - because it would be cunstructing strings on every call, and would be returning references to temporary objects.